addons/mail/static/src/models/attachment/attachment.js

```diff
@@ -109,7 +109,11 @@ function factory(dependencies) {
                 return `/web/image/${this.id}?signature=${this.checksum}`;
             }
             if (this.isPdf) {
-                return `/web/static/lib/pdfjs/web/viewer.html?file=/web/content/${this.id}`;
+                if (!this.accessToken && this.originThread && this.originThread.model === 'mail.channel') {
+                    return `/web/static/lib/pdfjs/web/viewer.html?file=/mail/channel/${this.originThread.id}/attachment/${this.id}`;
+                }
+                const accessToken = this.accessToken ? `?access_token%3D${this.accessToken}` : '';
```

Comment:
Any specific reason to use `%3D` instead of `=`? It would be more clear with `=`.

Reply:
Hello @seb-odoo i tried with using '=' but we can't get access token parameter on controller side because url is not parsing correctly.